doc/dev/CustomRepoGuide.md

```diff
@@ -0,0 +1,387 @@
+
+~~~yml
+script:
+    - docker exec dock cmake -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_CXX_FLAGS="$COVERAGE $WERROR" ../source
+    - docker exec dock make -j2
+    - docker exec dock ctest
+~~~
+ 
+The command `docker exec dock` just sends a bash command to the docker image that we named "dock". The only noteworthy thing is the cmake call, where we pass our environmental variables which we set in the `matrix:` section to cmake. Additionally you should have a look at the end of the cmake call: `../source`. Here we specify the path to our source files, which we mounted to "/home/nuto/source". Since we are using ".." our current location inside the image is obviously a folder located inside "/home/nuto". This folder is "build". This is the predefined starting directory of the current nuto image. You have to be careful which command you call from which directory. Calling a command from the wrong directory might cause your tests to fail. You can always check what you are doing inside the docker container by pulling the image to your computer and copying the commands from your .travis.yml. This will be discussed in the following section.
+
+
+
+**Additional notes**
+
+The travis .travis.yml can contain additional section that are not needed here. You can find an overview here: 
+
+https://docs.travis-ci.com/user/customizing-the-build/
+
+If you make any changes to your `.travis.yml` and want to know if it is valid before pushing to your repository, you can copy and paste the code into the corresponding field on 
+
+http://lint.travis-ci.org/
+
+Or follow the instructions here:
+
+https://docs.travis-ci.com/user/travis-lint
+
+# Docker
+
+Docker is a virtual machine that we use to run an image on travis. This way we avoid installing all the necessary components for NuTo over and over again during each test run on travis. This saves a lot of time but adds a little bit more complexity to the whole test environment. However, if your test fails and the reason is not obvious due to the output, you can easily reproduce the travis results on your own computer in order to find the exact issue. First you need to install docker
+
+~~~bash
+sudo apt-get install docker
+~~~
+
+Now you can simply copy the commands from the .travis.yml. Watch out that you might need to replace the `$(pwd)` with the location of the NuTo sources on your computer when starting the image. The results should be the same as on Travis. You can start modifying/adding/removing commands and see how you get your tests to pass. If you need to see what is actually going on inside the image, just replace the `itd` with `-it` and drop the `--name dock` when starting the image. This will send you to the bash inside of the image so that you can navigate through the folders etc. a little bit more comfortable. You can also drob `docker exec dock` in front of every command. To get back to your system just type
```

Comment:
Why do I have to "drop the `--name dock`"? Also "drob"->"drop" a couple sentences later.